components/camel-mina2/src/main/java/org/apache/camel/component/mina2/Mina2Producer.java

```diff
@@ -65,6 +65,7 @@
 public class Mina2Producer extends DefaultProducer implements ServicePoolAware {
 
     private static final Logger LOG = LoggerFactory.getLogger(Mina2Producer.class);
+    private final ResponseHandler handler;
     private IoSession session;
     private CountDownLatch latch;
```

Comment:
Maybe rename this to responseLatch (or something)? To avoid confusion about what the latches are used for.